routes/referenda.js

```diff
@@ -0,0 +1,190 @@
+
+  res.render('referenda/referendum.pug', { req, referendum, isActive, canVote, candidates, resultsVisible, votes })
+})
+
+app.post(__l('/referendum/:slug'), function (req, res) {
+  const data = referenda.load()
+  const referendum = data.find(ref => ref.slug === req.params.slug)
+  if (referendum === undefined) {
+    return res.status(404).render('error.pug', {req, err: 'No such referendum'})
+  }
+  const participants = referendum.participants || []
+  const isActive = isReferendumActive(referendum)
+  const hasVoted = participants.includes(req.user)
+  const canVote = isActive && !hasVoted
+  if (!canVote) {
+    return res.status(403).render('error.pug', {req, err: 'Cannot vote in this referendum'})
+  }
+  const candidates = referendum.candidates || []
+
+  const votes = candidates.reduce((votes, candidate) => {
+    const userVote = req.body[candidate] === 'on' ? 1 : 0
+    votes[candidate] = (votes.hasOwnProperty(candidate) ? votes[candidate] : 0) + userVote
+    return votes
+  }, referendum.votes || {})
+  participants.push(req.user)
+
+  referendum.votes = votes
+  referendum.participants = participants
+
+  referenda.save(data)
+
+  res.redirect(`/referendum/${referendum.slug}`)
+})
+
+app.get(__l('/referendum/:slug/edit'), function (req, res) {
+  const data = referenda.load()
+  const referendum = data.find(ref => ref.slug === req.params.slug)
+  if (referendum === undefined) {
+    return res.status(404).render('error.pug', {req, err: 'No such referendum'})
+  }
+  const isActive = isReferendumActive(referendum)
+  const isOwner = referendum.createdBy === req.user
+  const canEdit = isActive && isOwner && !referendum.published
+  if (!canEdit) {
+    return res.status(403).render('error.pug', {req, err: 'Referendum cannot be edited'})
+  }
+  const candidates = referendum.candidates || []
+
+  res.render('referenda/referendum-edit.pug', { req, referendum, candidates })
+})
+
+app.post(__l('/referendum/:slug/edit'), function (req, res) {
+  const data = referenda.load()
+  const referendumIndex = data.findIndex(ref => ref.slug === req.params.slug)
+  const referendum = data[referendumIndex]
+  if (referendum === undefined) {
+    return res.status(404).render('error.pug', {req, err: 'No such referendum'})
+  }
+
+  const isActive = isReferendumActive(referendum)
+  const isOwner = referendum.createdBy === req.user
+  const canEdit = isActive && isOwner && !referendum.published
+  if (!canEdit) {
+    return res.status(403).render('error.pug', {req, err: 'Referendum cannot be edited'})
+  }
+  const candidates = referendum.candidates || []
+
+  if (req.body.name) {
+    candidates.push(req.body.name)
+    referendum.candidates = candidates
+  } else if (req.body.publish === 'on') {
+  } else {
+    return res.status(400).render('error.pug', {req, msg: __('referendum-create-fail')})
+  }
+
+  data[referendumIndex] = referendum
+  referenda.save(data)
+
+  res.redirect(`/referendum/${referendum.slug}/edit`)
```

Comment:
You can pass variables to the i18n stuff ([docs](https://www.npmjs.com/package/i18n#i18n__)), e.g.
```suggestion
  res.redirect(__l('/referendum/:slug/edit', {slug: referendum.slug})
```